test/lit/passes/gufa-refs.wast

```diff
@@ -6082,15 +6084,58 @@
   ;; CHECK-NEXT:   (unreachable)
   ;; CHECK-NEXT:  )
   ;; CHECK-NEXT: )
-  (func $test
+  (func $test-set-bottom
     ;; We should not error on sets to bottom types, even if they are cast from
     ;; valid values.
     (array.set $array
-     (ref.cast nullref
-       (global.get $global)
-     )
-     (i32.const 0)
-     (i32.const 0)
-   )
+      (ref.cast nullref
+        (global.get $global)
+      )
+      (i32.const 0)
+      (i32.const 0)
+    )
+  )
+
+  ;; CHECK:      (func $loop-tuple-br_on (type $2)
+  ;; CHECK-NEXT:  (tuple.drop 3
+  ;; CHECK-NEXT:   (loop $loop (type $0) (result i64 nullref i32)
+  ;; CHECK-NEXT:    (drop
+  ;; CHECK-NEXT:     (block
+  ;; CHECK-NEXT:      (drop
+  ;; CHECK-NEXT:       (br_on_null $loop
+  ;; CHECK-NEXT:        (ref.null none)
```

Comment:
Why has `any` changed to `none`? Does GUFA do that?

Reply:
This is actually something we do automatically all the time (I forgot if in refinalize or in parsing, but one of them). Basically we turn nulls into the most refined types that we can, of the bottom type, because that is always valid and more precise.